packages/core-flows/src/region/steps/update-regions.ts

```diff
@@ -0,0 +1,59 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  FilterableRegionProps,
+  IRegionModuleService,
+  UpdatableRegionFields,
+} from "@medusajs/types"
+import {
+  getSelectsAndRelationsFromObjectArray,
+  promiseAll,
+} from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type UpdateRegionsStepInput = {
+  selector: FilterableRegionProps
+  update: UpdatableRegionFields
+}
+
+export const updateRegionsStepId = "update-region"
+export const updateRegionsStep = createStep(
+  updateRegionsStepId,
+  async (data: UpdateRegionsStepInput, { container }) => {
+    const service = container.resolve<IRegionModuleService>(
+      ModuleRegistrationName.REGION
+    )
+
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray([
+      data.update,
+    ])
+
+    const prevData = await service.list(data.selector, {
+      select: selects,
+      relations,
+    })
+
+    const regions = await service.update(data.selector, data.update)
+
+    return new StepResponse(regions, prevData)
+  },
+  async (prevData, { container }) => {
+    if (!prevData?.length) {
+      return
+    }
+
+    const service = container.resolve<IRegionModuleService>(
+      ModuleRegistrationName.REGION
+    )
+
+    await promiseAll(
+      prevData.map(
+        async (region) =>
+          await service.update(region.id, {
```

Comment:
q: don't we have the ability to provide all the data such that we pass an array of [{ id, name, currency, meta }]?